server/src/test/java/org/elasticsearch/cluster/metadata/IndexMetadataVerifierTests.java

```diff
@@ -209,6 +255,69 @@ public void testReadOnlyVersionCompatibility() {
             );
             service.verifyIndexMetadata(idxMetadata, IndexVersions.MINIMUM_COMPATIBLE, IndexVersions.MINIMUM_READONLY_COMPATIBLE);
         }
+        {
+            var idxMetadata = newIndexMeta(
+                "searchable-snapshot-no-write-block",
+                Settings.builder()
+                    .put(IndexMetadata.SETTING_VERSION_CREATED, indexCreated)
+                    .put(INDEX_STORE_TYPE_SETTING.getKey(), SearchableSnapshotsSettings.SEARCHABLE_SNAPSHOT_STORE_TYPE)
+                    .build()
+            );
+            String message = expectThrows(
+                IllegalStateException.class,
+                () -> service.verifyIndexMetadata(idxMetadata, IndexVersions.MINIMUM_COMPATIBLE, IndexVersions.MINIMUM_READONLY_COMPATIBLE)
+            ).getMessage();
+            assertThat(
```

Comment:
Is this a new req that the searchable snapshot must be marked read-only?

Reply:
No, since searchable snapshots always have the write block. I agree it's confusing to have a unit test for a situation that is not supposed to happen, maybe I should just removed this?